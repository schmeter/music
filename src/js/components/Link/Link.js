import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';


const isExternal = (path) => /^https?:\/\/.+/.test(path);

class Link extends React.Component {
    render() {
        const { className, to, onClick, children } = this.props;
        const props = {
            className,
            onClick
        };
        const content = children || to;
        return isExternal(to) ? (
            <a
                target="_blank"
                href={to}
                {...props}
            >
                {content}
            </a>
        ) : (
            <NavLink
                exact
                activeClassName="active"
                to={to}
                {...props}
            >
                {content}
            </NavLink>
        );
    }
}

Link.propTypes = {
    to: PropTypes.string.isRequired,
    children: PropTypes.node,
    onClick: PropTypes.func,
    className: PropTypes.string
};

export default Link;
