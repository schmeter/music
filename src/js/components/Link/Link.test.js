import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Link from './Link';

describe('Link', () => {
  const mockedEvent = {
    defaultPrevented: false,
    preventDefault: jest.fn(),
  };
  const onClick = jest.fn();

  it('renders correctly with external url', () => {
    const props = {
      to: 'http://test',
      children: 'test',
      className: 'test',
      onClick,
    };
    const component = shallow(<Link {...props} />);

    component.simulate('click', mockedEvent);

    expect(toJson(component)).toMatchSnapshot();
    expect(props.onClick).toHaveBeenCalled();
  });

  it('renders correctly with internal url', () => {
    const props = {
      to: 'test',
      onClick,
    };
    const component = shallow(<Link {...props} />);

    component.simulate('click', mockedEvent);

    expect(toJson(component)).toMatchSnapshot();
    expect(props.onClick).toHaveBeenCalled();
  });
});
