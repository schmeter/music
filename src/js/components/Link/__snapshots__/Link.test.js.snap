// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly with external url 1`] = `
<a
  className="test"
  href="http://test"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "defaultPrevented": false,
            "preventDefault": [MockFunction],
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  rel="noopener noreferrer"
  target="_blank"
>
  test
</a>
`;

exports[`Link renders correctly with internal url 1`] = `
<NavLink
  activeClassName="active"
  exact={true}
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "defaultPrevented": false,
            "preventDefault": [MockFunction],
          },
        ],
        Array [
          Object {
            "defaultPrevented": false,
            "preventDefault": [MockFunction],
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  to="test"
>
  test
</NavLink>
`;
